src/isar/state_machine/state_machine.py

```diff
@@ -92,30 +180,106 @@ def __init__(
         self.transitions_log_length = transitions_log_length
         self.transitions_list: Deque[States] = deque([], self.transitions_log_length)
 
+    #################################################################################
+    # Transition Callbacks
+    def _step_initiated(self) -> None:
+        self.current_step.status = StepStatus.InProgress
+        self.publish_step_status()
+        self.logger.info(
+            f"Successfully initiated "
+            f"{type(self.current_step).__name__} "
+            f"step: {str(self.current_step.id)[:8]}"
+        )
+
+    def _pause(self) -> None:
+        self.paused = True
+
+    def _resume(self) -> None:
+        self.logger.info("Resuming mission!")
+        self.paused = False
+        self.current_mission.status = MissionStatus.InProgress
+        self.current_task.status = TaskStatus.InProgress
+        self.publish_mission_status()
+        self.publish_task_status()
+        self.queues.resume_mission.output.put(True)
+        self.current_task.reset_task()
+        self.update_current_task()
+        self.update_current_step()
+
+    def _finalize(self) -> None:
+        self.publish_mission_status()
+        self.log_step_overview(mission=self.current_mission)
+        self.reset_state_machine()
+        state_transitions: str = ", ".join(
+            [
+                f"\n  {transition}" if (i + 1) % 10 == 0 else f"{transition}"
+                for i, transition in enumerate(list(self.transitions_list))
+            ]
+        )
+        self.logger.info(f"State transitions:\n  {state_transitions}")
+
+    def _mission_started(self) -> None:
+        self.logger.info(f"Starting new mission: {self.current_mission.id}")
+        self.current_mission.status = MissionStatus.InProgress
+        self.current_task.status = TaskStatus.InProgress
+        self.publish_mission_status()
+        self.publish_task_status()
+        self.log_step_overview(mission=self.current_mission)
+        self.update_current_task()
+        self.update_current_step()
+
+    def _step_finished(self) -> None:
+        self.publish_task_status()
```

Comment:
Should the task status be published here? There is not necessarily a change in the task status and if there is a change it will be published in `update_current_task`

Reply:
Meant to be `self.publish_step_status()`